e2e_playwright/st_tabs_test.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+def test_tabs_render_correctly(themed_app: Page, assert_snapshot: ImageCompareFunction):
+    st_tabs = themed_app.get_by_test_id("stTabs")
+    expect(st_tabs).to_have_count(3)
+
+    assert_snapshot(st_tabs.nth(0), name="st_tabs-sidebar")
+    assert_snapshot(st_tabs.nth(1), name="st_tabs-text_input")
+    assert_snapshot(st_tabs.nth(2), name="st_tabs-date_input")
+
+
+def test_displays_correctly_in_sidebar(app: Page):
+    expect(
+        app.get_by_test_id("stSidebar").locator('.stTabs [data-baseweb="tab"]')
+    ).to_have_count(2)
+    expect(app.get_by_text("I am in the sidebar")).to_have_count(1)
+    expect(app.get_by_text("I am in the sidebarI'm also in the sidebar")).to_have_count(
+        1
+    )
+
+
+def test_contains_all_tabs_when_overflowing(app: Page):
+    expect(
+        app.get_by_test_id("stExpander").locator('.stTabs [data-baseweb="tab"]')
```

Comment:
nit: I think it is a bit more future-proof to use: `get_expander(app, "Expander")` here for this